code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+
+## Inline queries
+
+Inline queries are sql queries in a file of another programming language, such
+as a PHP file with sql queries.
+
+### Queries in php files
+
+Queries SHOULD be in [Nowdoc][php nowdoc format] format and MAY use
+[Heredoc][php heredoc format]. The opening tag SHOULD be `<<<'SQL'` for Nowdoc 
+or `<<<SQL` for Heredoc.
+
+The query itself SHOULD NOT be indented. 
+
+Good example
+
+```php
+<?php
+class Foo
+{
+    public function bar()
+    {
+        $query = <<<'SQL'
+SELECT id
+FROM my_table
+WHERE id > 10;
+SQL;
+    }
+}
+```
+
+Bad example:
+
+```php
+<?php
+class Foo
+{
+    public function bar()
+    {
+        $query = <<<'SQL'
+        SELECT id
+        FROM my_table
+        WHERE id > 10;
+SQL;
+
+        $query = "SELECT id
+        FROM my_table
+        WHERE id > 10;";
+
+        $query = '
+SELECT id
+FROM my_table
+WHERE id > 10;';
+    }
+}
+```
+
+## Database design
+
+### Tables
+
+Table names
+
+- MUST be in English
+- MUST be in lower snake case
+- MUST be singular
+
+Tables
+
+- MUST use the `utf8` character set.
+- MUST have an `id` column
+- whose sole purpose is to link two (or more) other tables MUST NOT have an `id` column
+- SHOULD have a `name` column
+
+It is RECOMMENDED to have tables for all entities. This includes entities like
+`color`. Queries SHOULD return the name, which can then be used in translations.
+
+Good table names:
+
+```txt
+model
+vehicle_type
+color
+```
+
+Bad table names:
+
+```txt
+voertuig_type
+models
+vehicleType
+```
+
+### Columns
+
+- Column names MUST be in English
+
+- Column names MUST be in lower snake case
+
+- Column names MUST be singular
+
+- Columns for possibly undefined values MUST be nullable
+
+- Columns for required values MUST NOT be nullable
+
+- Columns MUST use the correct type for the information they need to store
+
+- Columns MAY have a default value. If a nullable column has a default value,
+  this default value SHOULD be `null`.
+
+The `id` column
+
+- MUST be named `id`
+- MUST be unsigned
+- SHOULD be of type integer
+- SHOULD be the first column in a table definition
+
+Good column names:
+
+```txt
+id
+vehicle_type_id
+```
+
+Bad column names:
+
+```txt
+ID
+vehicleTypeId
+```
+
+### Foreign keys
+
+Columns that reference other tables MUST have a foreign key relation to that
+other table. The name of the column SHOULD be of the form `<referenced_table_name>_id`
+
+## Formatting
+
+- Keywords must be written in upper case
+- MySQL function names are considered keywords
+- Table names and column names MUST be written as they are defined, in lower case
+- Indents MUST be two spaces
+- Keywords MUST be followed by exactly one space
+- Keywords that consist of multiple words MUST be written on one line
+- Queries SHOULD be parametrized statements
+- Parameters SHOULD be specified by name
+- Operators MUST be surrounded by one space
+- Table names, column names and database names SHOULD NOT be surrounded with backticks
```

Comment:
Doesn't this pose problems for columns that are named a [reserved word](https://dev.mysql.com/doc/refman/5.7/en/keywords.html)? (Or is this effect negated by using lowercase?)

I am aware backticks are not part of the ANSI SQL standard. If this is the reason to avoid them I would like that to be added...